src/types.rs

```diff
@@ -44,25 +49,76 @@ pub struct Ctip {
     pub value: Amount,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct Sidechain {
-    pub sidechain_number: SidechainNumber,
-    pub data: Vec<u8>,
-    pub vote_count: u16,
-    pub proposal_height: u32,
-    pub activation_height: u32,
+#[derive(BorshSerialize, Clone, Debug, Deserialize, Display, Eq, PartialEq, Serialize)]
+#[display("{}", hex::encode(_0))]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct SidechainDescription(pub Vec<u8>);
+
+impl SidechainDescription {
+    pub fn sha256d_hash(&self) -> [u8; 32] {
+        let bytes = borsh::to_vec(self).unwrap();
```

Comment:
`borsh` is an encoding library for *canonical* encodings, typically used for hashing. It might make more sense to use `bitcoin::consensus::Encodable` instead here